tests/bluetooth/common/testlib/src/timeout.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cmdline.h"  /* native_posix command line options header */
+#include "bs_types.h" /* For `bs_time_t` */
+#include "posix_native_task.h"
+
+static bs_time_t timeout_us;
+
+bs_time_t timeout_get(void)
+{
+	return timeout_us;
+}
+
+static void add_timeout_arg(void)
+{
+	static struct args_struct_t timeout_opts[] = {
+		{
+			.dest = &timeout_us,
+			.type = 'I',
```

Comment:
U

Reply:
Actually it would be nicer if you took it in as a double, and then converted in code to bs_time_t.
(you can just have the command line parsing put the value in a intermediate double variable and do 
``` 
static void sim_length_found(char * argv, int offset){
  timeout_us = sim_length;
  bs_trace_raw(9,"cmdarg: sim_length set to %"PRItime"\n", args_g->sim_length);
}
```

Reply:
we can fix that in another PR

Reply:
Ok, but please remember to do it. Users will be annoyed/confused about why the phy takes happily 30e6 as parameter, while this one only takes 30000000

Reply:
The current code works fine with `5000000001` even.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/68428270/c172badb-f4c8-4ead-a553-3dbc1a1e0592)